cli/src/commands/rebase.rs

```diff
@@ -268,11 +268,12 @@ pub(crate) fn cmd_rebase(
             &rebase_options,
         )?;
     } else {
-        let destination = args
-            .destination
-            .destination
-            .as_ref()
-            .expect("clap should enforce -d when used with -b");
+        let Some(destination) = args.destination.destination.as_ref() else {
+            return Err(cli_error(
+                "jj rebase without `-r`, `-s`, or `-b` requires the `--destination <DESTINATION>` \
```

Comment:
nit: `--destination` is required even if `-b` is explicitly specified. Maybe we can say "--destination is required to rebase a whole branch" or "--insert-before/after is not supported when rebasing a whole branch"?

We can also remove `conflicts_with` from `--branch`.